tests/lib/c_lib/src/main.c

```diff
@@ -299,15 +322,21 @@ void test_strxspn(void)
 void test_memcmp(void)
 {
 	int ret;
-	unsigned char m1[5] = { 1, 2, 3, 4, 5 };
-	unsigned char m2[5] = { 1, 2, 3, 4, 6 };
+	unsigned char m1[7] = "123457";
```

Comment:
You can use:

    unsigned char m1[] = "12345A";

to avoid the potential inconsistency between the array declared size and the required size for the initial value.

Also I propose changing the suffix characters from digits to letters to make it more clear that you're not intending this to be a 7 character long sequence.

Reply:
ok, I agree with you , I will update it.